app/views/play/cinematic/DialogSystem.js

```diff
@@ -0,0 +1,314 @@
+    svg.setAttribute('xmlns', SVGNS)
+    svg.setAttribute('height', `${height}`)
+    svg.setAttribute('width', `${width}`)
+
+    svg.style.pointerEvents = div.style.pointerEvents = 'none'
+    svg.style.position = div.style.postition = 'absolute'
+
+    canvasDiv.appendChild(svg)
+    canvasDiv.appendChild(div)
+
+    this.shownDialogBubbles = []
+    this._context = {}
+  }
+
+  /**
+   * This context object can be accessed from the dialog text templates.
+   * E.g. a context object of: `{ name: 'Mary' }` can then be used in the following
+   * dialog text:
+   *   `Hello, <%=o.name%>`
+   * Which then appears as:
+   *   `Hello, Mary`
+   *
+   * @param {Object} context - used for interpolation.
+   */
+  set templateContext (context) {
+    this._context = context
+  }
+
+  /**
+   * The system method that is run on every dialogNode.
+   * @param {import('../../../schemas/selectors/cinematic').DialogNode} dialogNode
+   * @returns {AbstractCommand[]}
+   */
+  parseDialogNode (dialogNode) {
+    const commands = []
+    const text = processText(dialogNode, this._context)
+    const shouldClear = getClearText(dialogNode)
+    const { x, y } = getTextPosition(dialogNode) || { x: 200, y: 200 }
+    const side = getSpeaker(dialogNode) || 'left'
+
+    if (shouldClear) {
+      commands.push(this.clearShownDialogBubbles())
+    }
+
+    if (text) {
+      commands.push((new SpeechBubble({
+        div: this.div,
+        svg: this.svg,
+        htmlString: text,
+        x,
+        y,
+        shownDialogBubbles: this.shownDialogBubbles,
+        side
+      })).createBubbleCommand())
+    }
+    return commands
+  }
+
+  /**
+   * @returns {AbstractCommand}
+   */
+  clearShownDialogBubbles () {
+    return new SyncFunction(() => {
+      this.shownDialogBubbles.forEach(el => el.remove())
+    })
+  }
+}
+
+let _id = 0
+/**
+ * Creates a speech bubble eagerly.
+ * Can return a command to display the speech bubble when called.
+ *
+ * Attaches itself to the svg canvas and div canvas.
+ */
+class SpeechBubble {
+  constructor ({
+    div,
+    svg,
+    htmlString,
+    x,
+    y,
+    shownDialogBubbles,
+    side
+  }) {
+    this.id = `speech-${_id++}`
+    const parser = new DOMParser()
+    const html = parser.parseFromString(htmlString, 'text/html')
+    const textDiv = html.body.firstChild
+    textDiv.style.display = 'inline-block'
+    textDiv.style.position = 'absolute'
+    textDiv.style.left = `${x}`
+    textDiv.style.top = `${y}`
+    textDiv.id = this.id
+
+    div.appendChild(textDiv)
+
+    // We've created and attached the dialog text.
+    // Now we can calculate the bounding box and draw the svg shape.
+    const bbox = textDiv.getBoundingClientRect()
+    const width = (bbox.right - bbox.left) + 2 * padding
+    const height = (bbox.bottom - bbox.top) + 2 * padding
+
+    const svgGroup = createSvgShape({
+      x,
+      y: y - height,
+      width,
+      height,
+      clazz: this.id,
```

Comment:
Should it be `class` ?

Reply:
Yeah this was some bad naming. `class` is a reserved keyword so I can't use that. `clazz` is not an easy name. It's flat out confusing. Great pickup.

Reply:
Oh i see, can we use `className` in that case.

Reply:
That's such a better name.